tests/functional/karate_tests/README.md

```diff
@@ -0,0 +1,49 @@
+# openwhisk-karate
+This repository contains all the Test Cases needed to do various forms of automated test on the OW environments.These tests will also complement the existing Scala based Test cases.
+These tests are based on the BDD model and enabled quick on-boarding of the tests.
+It is based of karate (https://github.com/intuit/karate) framework. 
+
+
+### How to run functional test
+1. Clone/Download the repo
+2. Navigate to the root folder
+2. Use the following command to run the above selected suite: `mvn test -Dtest="com.karate.openwhisk.smoketests.SmokeTestRunner"` (This will run all the tests in com.karate.openwhisk.smoketests package.)
+
+### How to run the performance test 
+1. Navigate to the root path.Check out the performance test package (/ow-karate/src/test/java/com/karate/openwhisk/performance)
+2. setUp(createActionTest.inject(rampUsers(5) over (5 seconds))
+    ).maxDuration(1 minutes).assertions(global.responseTime.mean.lt(1100))
+Tweak the above values as per your test needs/load requirements
+3. Use the following command to run the performance: 
+`mvn clean install gatling:test -Dgatling.simulationClass=com.karate.openwhisk.performance.LoadTest`
+
+### How to add more tests
+
+1. Select a package(Type of test).Example Smoke test
+2. Add a new feature file which has your test with the following tags `@smoke`
+
+### How to add a new test type
+1. Create a package in `src/test/java`.
+2. We can create test types like regression, sanity etc.For example `com.karate.openwhisk.sanitytests`
+3. Create a feature and runner file inside the above package 
```

Comment:
Is there a reason that java and non-java source files need to be organized in the same dir? This is not common, typically non-sourcecode is kept in a "resources" dir, or a source dir specific to the language.